core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -519,10 +558,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
     scheduler.sc.env.blockManager.master.decommissionBlockManagers(executorsToDecommission)
 
     if (!triggeredByExecutor) {
-      executorsToDecommission.foreach { executorId =>
-        logInfo(s"Notify executor $executorId to decommissioning.")
-        executorDataMap(executorId).executorEndpoint.send(DecommissionExecutor)
-      }
```

Comment:
I wonder do you face a real network issue on this?

Reply:
@Ngone51 Network overload will come from simultaneous block migration if we decommission large chunk of executors at once. If we do it in batches we limit migration of blocks at once.

Reply:
Of course, I know the network can suffer the overload when there's a large chunk of decommissioning executors at the same time.

My question is, did you see the issue happen in a real cluster? AFAIK, for example, [the average frequency of spot instance interruption in AWS](https://aws.amazon.com/ec2/spot/instance-advisor/) is only 5%, which I think is a low probability that could lead to plenty of executors to decommission in a short time.

Reply:
No, i haven't seen this in a public cloud. We have some experience with this issue in private cloud which had bigger timeout for forceful node removal and this was the motive of this PR to give some control of decommissioning process to user.

Reply:
What's the symptom when the issue happens? Could executors detect the issue themselves? I'm thinking of whether we could control the migration on the executor side.

Reply:
I would also love more information if your able to share @q2w about the issues you ran into in production.